### PR TITLE
bugfix: release push tag at main branch

### LINKED ISSUE #21: bug: Release at a detached commit
## Description
Release tag push at a detached commit of a fork of `depandabot`.

Which lead to `v0.2.4` as the only available last tag and changelog contains whole history.

## Example:
image

image

detached commit: https://github.com/tensorchord/envd-lsp/commit/9bb022fd9674a11aaa535eff9791246b185ae543

## Reproduction

```shell
git tag
v0.0.1-alpha.1
v0.0.1-alpha.2
v0.0.1-alpha.3
v0.2.0
v0.2.2
v0.2.3
v0.2.4
v0.2.5
v0.3.0
v0.3.1
v0.3.5

git describe --abbrev=0 --tags
v0.2.4
```

## Additional Info
Action: https://github.com/tensorchord/envd-lsp/actions/runs/3828232773/jobs/6513594739
The Action checkout the detached branch.

image

## Fix
To fix it, we need to:
- Remove `mathieudutour/github-tag-action@v6.1` and turn to `git tag push`
- Add `ref: main` to `actions/checkout@v3` to force pull main branch
- Add `kachick/wait-other-jobs@v1` to wait PR merged finished before push tag

```
- name: Wait other jobs(CI and lint)
      uses: kachick/wait-other-jobs@v1
      timeout-minutes: 10
      with:
        min-interval-seconds: '10'
        github-token: '${{ secrets.GITHUB_TOKEN }}'
```
---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by cutecutecat**
close #21 

Example:
A correct release at the main branch
https://github.com/cutecs/envd-lsp/releases/tag/v0.3.5

Signed-off-by: cutecutecat <starkind1997@gmail.com>

### CODE REVIEW

File: .github/workflows/dep_trig_release.yml [hunk 1]
```diff
@@ -36,24 +36,35 @@ jobs:
       env:
         PR_URL: ${{ github.event.pull_request.html_url }}
         GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Wait for status checks
+      uses: WyriHaximus/github-action-wait-for-status@v1.7.1
+      with:
+        ignoreActions: "Merge a dependabot PR"
+        checkInterval: 10
+      env:
+        GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (kemingy) [inline .github/workflows/dep_trig_release.yml:45]: Why is this required?

Comment (cutecutecat) [reply .github/workflows/dep_trig_release.yml:45]: `gh pr merge --auto --squash "$PR_URL"` will not block Action, while the PR merge will wait until status check ends. Next step `push tag` will continue immediately after this command.

It we don't wait before push tag, then the tag will be pushed at previous(HEAD^1) commit.

@kemingy

Comment (kemingy) [reply .github/workflows/dep_trig_release.yml:45]: So the current step is:

1. dependabot upgrade envd
2. trigger CI lint/test/build and auto merge + push tag
3. new tag trigger release (by action `on.push.tags`) but push tag also trigger release (by `uses release.yml`)

Comment (cutecutecat) [reply .github/workflows/dep_trig_release.yml:45]: > So the current step is:
> 
> 1. dependabot upgrade envd
> 2. trigger CI lint/test/build and auto merge + push tag
> 3. new tag trigger release (by action `on.push.tags`) but push tag also trigger release (by `uses release.yml`)

`new tag trigger release` only happens when user push a tag, if the tag is pushed by Action, it will not be triggered, so I turned to last one to straightly triger it in previous commit.

If we don't need to tag and release some special version of `envd-lsp` not according to `envd` manually, the first one could be removed. @kemingy

**Review by kemingy — APPROVED** [review]
/lgtm
